src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx

```diff
@@ -1,21 +1,50 @@
-import React from "react";
-import { useSelector } from "react-redux";
+import React, { useState, useEffect } from "react";
 import { TitleView } from "../TitleView";
 import { BLinfo } from "../TitleView/BLinfo";
+import { isERC721Token, getTokenOwner } from "./../../../services/token";
+import { getLogger } from "./../../../utils/logger";
+import { useEscrowContractUsers } from "../../../hooks";
+const { trace, error } = getLogger("Component:DetailedCertificateVerifyBlock");
 
-export const TitleTransferPanelContent = () => {
-  const { holderAddress, beneficiaryAddress } = useSelector((state: any) => ({
-    holderAddress: state.token.holderAddress,
-    beneficiaryAddress: state.token.beneficiaryAddress
-  }));
+interface TitleTransferPanelContentProps {
+  document: object;
+}
+
+export const TitleTransferPanelContent = ({ document }: TitleTransferPanelContentProps) => {
+  const [tokenError, setError] = useState(null);
+  const [tokenOwner, setTokenOwner] = useState("");
+  const { holderAddress, beneficiaryAddress } = useEscrowContractUsers({ escrowContractAddress: tokenOwner });
+
+  const isToken = isERC721Token(document);
+  useEffect(() => {
+    async function fetchTokenOwner() {
+      try {
+        const owner = await getTokenOwner({ document });
+        trace(`Token Owner: ${owner}`);
+        setTokenOwner(owner);
+      } catch (e) {
+        error(`error in fetching token owner: ${JSON.stringify(e)}`);
+        setError(e.message);
+      }
+    }
+    if (isToken) fetchTokenOwner();
+  }, [document, isToken]);
+
+  if (!isToken) return null;
```

Comment:
If it's not a token this panel wont even be displayed right? no need short circuit